jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -151,7 +126,10 @@ else if (urlString.startsWith("jar:"))
      */
     public static Resource newResource(String resource) throws IOException
     {
-        return newResource(resource, __defaultUseCaches);
+        URI uri = URI.create(resource);
+        FileSystem fileSystem = FileSystems.newFileSystem(uri, new HashMap<>());
```

Comment:
new HashMap????   What's wrong with `Path.of(uri)`?

Reply:
That's one question mark I have noted down. Here's why this `FileSystems.newFileSystem` call is needed:

it is because when you pass a URI that points to a filesystem that hasn't been 'mounted' yet (e.g.: `jar:file:/some/path/foo.jar!/another/path/foo.txt`) you must explicitly create the filesystem of that URI otherwise `Path.of(uri)` will throw `FileSystemNotFoundException`.

The extra problem is that once a filesystem is created, it must eventually be closed otherwise we leak. `Resource` currently implements `Closeable` but nowhere is that used in practice, and it would be very inconvenient to use everywhere. This is yet another shortcoming of the existing `Resource` API that has to be sorted.

Reply:
That's why I think we should have extra steps for "jar:" and "jar:file:" resources, as they need an extra file system created.   I think it is OK to have a single pool of these per JVM.  We could probably even have a sweeper that checked if the underlying file still existed with same timestamp and size... and delete the pooled filesystem if the file has changed.

Reply:
What kind of extra step do you have in mind for "jar:" and "jar:file:" resources?

Reply:
done.